src/app/pages/2_Sequence_Search.py

```diff
@@ -49,24 +46,55 @@
                 st.session_state[NCBI_SUMMARY_FORM] = True
                 st.session_state[NCBI_DF] = ncbi.get_data(database, search_term)
                 data_processing.format_ncbi_summary()
-
     df_aggrid = pd.DataFrame()
     if st.session_state[NCBI_SUMMARY_FORM]:
         aggrid_table = aggrid_table()  # Initialize and Render Aggrid Table
         df_aggrid = aggrid_table["data"]
-
     # Streamlit UI to filter the dataset
     df_filtered, applied_filters = filter_dataframe(df_aggrid)
     if st.session_state[NCBI_DF_FILTER]:
         st.dataframe(df_filtered)
         st.write(f"Size of final data: {len(df_filtered)}")
-
-        if st.button("Prepare Download"):
-            download_data(database, df_filtered["Id"].unique(), search_term)
-
-    # Streamlit UI to show the applied filters
-    show_filters_info(applied_filters)
-
+        # Streamlit UI to show the applied filters
+        show_filters_info(applied_filters)
+    # User has made a search query
+    if not df_aggrid.empty:
+        # Dropdown menu for selecting download format
+        download_format = st.selectbox(
+            "Select an option:", [CSV_DOWNLOAD, FASTA_DOWNLOAD]
+        )
+        # Option 1: Download the correct DataFrame as CSV
+        if download_format == CSV_DOWNLOAD:
+            # User has applied filters using `Add filters` checkbox
+            if applied_filters:
+                st.download_button(
+                    label="Download table as CSV",
+                    data=df_filtered.to_csv(index=False).encode("utf-8"),
+                    file_name="sequence_search_table.csv",
+                    mime="text/csv",
+                    key="df_filtered_csv_button",
+                    help="Click button to Download this table as a CSV file",
+                    args={"as_attachment": True},
+                )
+            # User has not applied filters using `Add filters` checkbox
+            else:
+                st.download_button(
+                    label="Download table as CSV",
+                    data=df_aggrid.to_csv(index=False).encode("utf-8"),
+                    file_name="sequence_search_table.csv",
+                    mime="text/csv",
+                    key="df_aggrid_csv_button",
+                    help="Click button to Download this table as a CSV file",
+                    args={"as_attachment": True},
+                )
+        # Option 2: Download the correct DataFrame as FASTA
+        elif download_format == FASTA_DOWNLOAD:
+            # User has applied filters using `Add filters` checkbox
+            if applied_filters:
+                download_data(database, df_filtered["Id"].unique(), search_term)
+            # User has not applied filters using `Add filters` checkbox
+            else:
+                download_data(database, df_aggrid["Id"].unique(), search_term)
```

Comment:
Can reduce code duplication, e.g.
```suggestion
         df_download = df_filtered if applied_filters else df_aggrid
        # Option 1: Download the correct DataFrame as CSV
        if download_format == CSV_DOWNLOAD:
              st.download_button(
                  label="Download table as CSV",
                  data=df_download.to_csv(index=False).encode("utf-8"),
                  file_name="sequence_search_table.csv",
                  mime="text/csv",
                  help="Click button to Download this table as a CSV file",
                  args={"as_attachment": True},
              )
        # Option 2: Download the correct DataFrame as FASTA
        elif download_format == FASTA_DOWNLOAD:
            download_data(database, df_download["Id"].unique(), search_term)
```